packages/client/hmi-client/src/components/sidebar-panel/artifact-list.vue

```diff
@@ -0,0 +1,81 @@
+<script setup lang="ts">
+import IconClose32 from '@carbon/icons-vue/es/close/32';
+
+defineProps<{
+	artifacts: { id: string | number; displayName: string }[];
+	selectedArtifactId?: string | number;
+}>();
+
+const emit = defineEmits<{
+	(e: 'artifact-clicked', id: string | number): void;
```

Comment:
I thought that the default `click` event is used for clicking the parent component in general, e.g. between elements or in the whitespace at the top/bottom of the list.

This event is fired when one of the children is clicked, so I thought it would be better to use a separate event.